web/packages/design/src/DataTable/InputSearch/InputSearch.tsx

```diff
@@ -34,22 +34,30 @@ export default function InputSearch({
   children,
   bigInputSize = false,
 }: Props) {
+  function submitSearch(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault(); // prevent form default
+    const { searchValue } = e.target as typeof e.target & {
+      searchValue: { value: string };
+    };
+
+    setSearchValue(searchValue.value);
```

Comment:
I think there's too much stuff happening and we obscure too much how this works.

`e.target` points to a `HTMLFormElement` of the form. `e.target.searchValue` points to the search input DOM node because [that's how forms in DOM work](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement):

> Named inputs are added to their owner form instance as properties, and can overwrite native properties if they share the same name (e.g. a form with an input named action will have its action property return that input instead of the form's action HTML attribute).

And then `searchValue.value` reads that data from the DOM node.

To add to that, casting `e.target` to a specific "made-up" type is not necessarily wrong, but we don't have to do that. `searchValue` exists on `e.target` only because of that implicit browser behavior described above. In such cases, we can get away with doing `e.target["searchValue"]` instead. It gives us the same type-level guarantees as saying `e.target as typeof e.target & { searchValue: …`, which is just us saying "Trust me, `searchValue` is really there on this object I swear 🤞".

---

What we could do instead is use `FormData` and through this also signal that we're depending on some implicit browser behavior.

```diff
diff --git a/web/packages/design/src/DataTable/InputSearch/InputSearch.tsx b/web/packages/design/src/DataTable/InputSearch/InputSearch.tsx
index 817ba47369..d4f15b271d 100644
--- a/web/packages/design/src/DataTable/InputSearch/InputSearch.tsx
+++ b/web/packages/design/src/DataTable/InputSearch/InputSearch.tsx
@@ -28,6 +28,8 @@ import {
   ColorProps,
 } from 'design/system';
 
+const searchInputName = 'searchValue';
+
 export default function InputSearch({
   searchValue,
   setSearchValue,
@@ -36,11 +38,10 @@ export default function InputSearch({
 }: Props) {
   function submitSearch(e: FormEvent<HTMLFormElement>) {
     e.preventDefault(); // prevent form default
-    const { searchValue } = e.target as typeof e.target & {
-      searchValue: { value: string };
-    };
+    const formData = new FormData(e.currentTarget);
+    const searchValue = formData.get(searchInputName) as string;
 
-    setSearchValue(searchValue.value);
+    setSearchValue(searchValue);
   }
 
   return (
@@ -51,7 +52,7 @@ export default function InputSearch({
           placeholder="Search..."
           px={3}
           defaultValue={searchValue}
-          name="searchValue"
+          name={searchInputName}
           autoFocus
         />
         <ChildWrapperBackground>

```

Admittedly, we still have to do `formData.get(searchInputName) as string`, because technically `FormData.prototype.get` can return either a string or a file (if input's type was set to `file`). But we clearly show that we depend on the values from the DOM and through the `searchInputName` we make sure that we always read data from the correct input.